### ISSUE TITLE
PYTHONPATH isn't expanded like how python handles it

### ISSUE THREAD

**Issue by frankli0324** [microsoft/pylance-release#3823]
Type: <b>Bug</b>

# Behaviour

set `PYTHONPATH=~/some_path` in ${workspaceRoot}/.env and add the settings like what https://code.visualstudio.com/docs/python/environments#_use-of-the-pythonpath-variable described

python accepts the path and finds the module under `/Users/myuser/some_path` at runtime

## Expected vs. Actual

vscode should resolve the path by `expanduser`ing the path, which isn't happening now

![image](https://user-images.githubusercontent.com/20221896/211727982-569cca4b-b867-4022-8a92-98badf28d68c.png)
![image](https://user-images.githubusercontent.com/20221896/211728056-3ce002b3-f792-4382-9be0-679d530cd62d.png)
![image](https://user-images.githubusercontent.com/20221896/211728075-862b2eef-7294-4adb-906e-09da4737bacd.png)
![image](https://user-images.githubusercontent.com/20221896/211728110-867eb0a3-1a4d-4bd5-8548-a5702c78914d.png)

# Diagnostic data

-   Python version (& distribution if applicable, e.g. Anaconda): 3.10.9
-   Type of virtual environment used (e.g. conda, venv, virtualenv, etc.): Global
-   Value of the `python.languageServer` setting: Pylance

<details>

<summary>Output for <code>Python</code> in the <code>Output</code> panel (<code>View</code>→<code>Output</code>, change the drop-down the upper-right of the <code>Output</code> panel to <code>Python</code>)
</summary>

<p>

```
XXX
```

</p>
</details>

<details>

<summary>User Settings</summary>

<p>

```

venvFolders: "<placeholder>"

languageServer: "Pylance"

linting
• pylintArgs: "<placeholder>"

formatting
• autopep8Args: "<placeholder>"

```

</p>
</details>

Extension version: 2022.20.1
VS Code version: Code 1.74.3 (Universal) (97dec172d3256f8ca4bfb2143f3f76b503ca0534, 2023-01-09T17:07:18.579Z)
OS version: Darwin x64 22.1.0
Modes: Unsupported
Sandboxed: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz (12 x 2600)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|2, 3, 3|
|Memory (System)|16.00GB (0.01GB free)|
|Process Argv|--crash-reporter-id 5332ccb7-5dae-43b3-a84e-84fd9b6df54f|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
vstes627:30244334
vslsvsres303:30308271
pythonvspyl392:30443607
vserr242cf:30382550
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
pythondataviewer:30285071
vscod805:30301674
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
cmake_vspar411:30581797
vsaa593cf:30376535
pythonvs932:30410667
cppdebug:30492333
vsclangdc:30486549
c4g48928:30535728
dsvsc012cf:30540253
azure-dev_surveyone:30548225
pyindex848:30577860
nodejswelcome1:30587005
2e4cg342:30602488
gswce1:30612156
3d0df643:30613357
dbltrim-noruby:30604474
89544117:30613380
fim-prod:30623723

```

</details>

**Comment by karthiknadig** [comment]
It looks like pylance is not processing the expansion of `~` if it is there in the env variable value.

**Comment by rchiodo** [comment]
This is similar to #3730. Should be easy to fix though.

**Comment by heejaechang** [comment]
This issue has been fixed in prerelease version 2023.1.31, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2023131-18-January-2023-PreRelease)
